cfgmgr/buffermgr.cpp

```diff
@@ -153,94 +152,99 @@ task_process_status BufferMgr::doSpeedUpdateTask(string port, bool admin_up)
     }
 
     speed = m_speedLookup[port];
-
-    string buffer_pg_key = port + m_cfgBufferPgTable.getTableNameSeparator() + LOSSLESS_PGS;
     // key format is pg_lossless_<speed>_<cable>_profile
     string buffer_profile_key = "pg_lossless_" + speed + "_" + cable + "_profile";
     string profile_ref = buffer_profile_key;
+    
+    vector<string> lossless_pgs = tokenize(pfc_enable, ',');
+    for (auto lossless_pg : lossless_pgs)
+    {
+        vector<FieldValueTuple> fvVectorPg, fvVectorProfile;
+        string buffer_pg_key = port + m_cfgBufferPgTable.getTableNameSeparator() + lossless_pg;
 
-    m_cfgBufferPgTable.get(buffer_pg_key, fvVectorPg);
+        m_cfgBufferPgTable.get(buffer_pg_key, fvVectorPg);
 
-    if (!admin_up && m_platform == "mellanox")
-    {
-        // Remove the entry in BUFFER_PG table if any
-        if (!fvVectorPg.empty())
+        if (!admin_up && m_platform == "mellanox")
```

Comment:
if this condition is true, as per the current logic, before exiting the if block, we return success. That is the right behavior. With the current logic, the code flow continues. Better to put the for loop within this if block for this case

Reply:
Updated. Now the for loop is moved to the if block. Thanks